### ISSUE TITLE
Enhancement: Add syntax highlighting for Python Engine

### ISSUE THREAD

**Issue by btel** [codenode/codenode#45]
one could use for example pygments to generate nicely highlighted Python code

**Comment by jamescasbon** [comment]
It _may_ be better to use javascript; otherwise syntax highlighting while typing will require a lot of network overhead.

Please see this thread: 
http://groups.google.com/group/codenode-devel/browse_frm/thread/3e0434593d772da6/720d5feb30822577?lnk=gst&q=highlighting#720d5feb30822577

**Comment by btel** [comment]
I agree, but it still may be easier to highlight the code only after the cell is evaluated.

**Comment by jamescasbon** [comment]
As of release 0.9a1, bespin supports multiple text areas.  The way to embed bespin (or other client side highlighter) is as follows.  Since Cell.js controls how a given input or output cell is created and changed, you need to change two points:

(docs: https://bespin.mozillalabs.com/docs/embedding/index.html )

Notebook.Cell.prototype.setType - this creates the widgets for the cell depending on the cell type.  For a cell of type 'input' you would need to change the current creation of a textarea and add a call to 'bespin.useBespin(textarea_id);' (you may also need to wrap the textarera in a div, since bespin adds dom elements after the textarea, and the cell code expects one element)

The second point is Notebook.Cell.prototype.content which acts as a getter or setter for the textarea content.  The api for getting and setting bespin content is here: 
https://bespin.mozillalabs.com/docs/pluginguide/editorapi.html

**Comment by jamescasbon** [comment]
PS I tried server side evaluation for my first markdown attempt.  However, the client side rendering was way better.  I think the javascript is harder to get started with but better i the long run, so I am trying to give you some jumping off points to start understanding it.

**Comment by btel** [comment]
Thanks for tips. I tried playing with bespin in codenode and I found that it multiple bespin editors were tricky when you configure them programmticaly, that is using bespin.useBespin function. There is a solution on the mailing list using jQuery, but I am not sure how to apply it:
http://groups.google.com/group/bespin/browse_thread/thread/7cf6c3711da703da

**Comment by btel** [comment]
PS I think that bespin may be too heavyweight for our simple cells. Perhaps, something like contenteditable div would be a better solution here. There is a nice example on the web: http://valums.com/edit-in-place/

**Comment by jamescasbon** [comment]
May be useful: 
http://en.wikipedia.org/wiki/Comparison_of_JavaScript-based_source_code_editors
